src/views/hooks/useEnforceScheduleLimit.tsx

```diff
@@ -0,0 +1,41 @@
+import useSchedules from '@views/hooks/useSchedules';
+import React, { useCallback } from 'react';
+
+import { Button } from '../components/common/Button';
+import { usePrompt } from '../components/common/DialogProvider/DialogProvider';
+
+const SCHEDULE_LIMIT = 10;
+
+/**
+ * Hook that checks if currently over the schedule limit
+ * @returns
```

Comment:
I'd give the function a more complete description

```suggestion
 * Hook that creates a function that enforces a maximum amount of schedules.
 *
 * If a new schedule can be created without exceeding the limit, the function returns true.
 * If not, it displays a prompt explaining the limit and returns false.
```

Reply:
Created a commit that addresses this.